lib/verse/http/spec/http_helper.rb

```diff
@@ -4,66 +4,32 @@ module Verse
   module Http
     module Spec
       module HttpHelper
-        class << self
-          attr_accessor :current_user
-
-          # Generate a new token for the given user.
-          # @param username [String] The name of the user. Must be registered via `add_user` method
-          # @return [String] The JWT token.
-          def new_token(username)
-            user = Verse::Spec.users.fetch(username.to_sym) {
-              raise "User `#{username}` not found. " \
-                    "Add it with `Verse::Http::Spec::HttpHelper.add_user(\"#{username}\", role, metadata, scopes)`" \
-                    "in your spec_helper.rb"
-            }
-
-            Verse::Http::Auth::Token.encode(
-              *user.values_at(:user_data, :role, :scopes),
-              exp: Time.now.to_i + 1_000_000
-            )
-          end
-        end
-
-        def current_user
-          Verse::Http::Spec::HttpHelper.current_user
-        end
-
-        def current_user=(username)
-          Verse::Http::Spec::HttpHelper.current_user = username
-        end
+        attr_reader :current_auth_context
 
         def app
           Verse::Http::Server
         end
 
-        def as_user(username)
-          old_user = Verse::Http::Spec::HttpHelper.current_user
-          Verse::Http::Spec::HttpHelper.current_user = username
-          yield
-        ensure
-          Verse::Http::Spec::HttpHelper.current_user = old_user
-        end
-
-        def current_user_context
-          current_user = Verse::Http::Spec::HttpHelper.current_user
-          return nil if current_user.nil?
+        def auth_token
+          raise "Missing current_auth_context" unless current_auth_context
```

Comment:
I think we could just return `nil` instead of an error, as in current_auth_context